src/main/java/org/llorllale/cactoos/matchers/TextMatcherEnvelope.java

```diff
@@ -0,0 +1,97 @@
+/*
```

Comment:
@dgroup why are we including this change here? Let's keep PRs focused. Feel free to create a new ticket requesting this envelope

Reply:
@llorllale  because this change allows avoiding code duplication within current PR.
I don't want to duplicate the code between EndsWith, StartsWith, TextIs, TextHasString.

Reply:
@dgroup you could've just implemented the envelope and left a puzzle for the rest. Let's keep to smaller PRs please

Reply:
and also the we need to duplicate the unit tests functionality.

Reply:
I'm too lazy to generate the duplication and then removing within one more pull request.
Even for money :).

Reply:
@llorllale, ok, will have to.

Reply:
@llorllale looks like it can't be generalized, will have to follow the current approach.

Reply:
I was wrong, done.

Reply:
@dgroup the merge has already begun...